### PR TITLE
boards/feather-m0: add support for feather-m0-lora version

### PR DESCRIPTION
**Pull request by tperale**
### Contribution description

The [feather-m0](https://www.adafruit.com/product/3178) board, lacked of support and documentation for its LoRa version that ship with a RFM95 module on it. The driver for the RFM95 LoRa module is exactly the same as the SX1276 already supported by RIOT-OS.
This PR add the pins mapping for the driver to the board definition to make the `feather-m0-lora` board work out of the box. I used those [schematics](https://cdn-learn.adafruit.com/assets/assets/000/032/914/original/feather_schem.png?1465421956) to find the pin mapping.

### Testing procedure

The contribution was tested on a `feather-m0-lora` with the `test/driver_sx127x` example.

### Issues/PRs references

I didn't found any reference to the `feather-m0-lora` in the existing issues.

### CODE REVIEW

**Comment by miri64** [comment]
There is already precedence set with the `feather-m0-wifi` board. Maybe the `feather-m0-lora` should be provided the same way to avoid confusion (create `feather-m0-lora` directory, copy most of `feather-m0-wifi` over, adapt for `lora` version).

**Comment by tperale** [comment]
@miri64 I created a separated directory as you suggested. Thank you

**Review by miri64 — APPROVED** [review]
ACK. I tested using a bare `feather-m0` (since I don't own the LoRa) version. Flashing `examples/gnrc_networking` or `examples/gnrc_lorawan` works as expected, but of course the application crashes whenever I try to access the network device (since my board does not have one connected). Please squash your commits. I suggest one commit for all modifications in `boards/` as an introductory commit for the `feather-m0-lora` board and one commit for all modifications in `tests/` (additions to `tests/*/Makefile.ci`; Murdock will reveal more here most likely).

File: boards/feather-m0-lora/Makefile.dep [hunk 1]
```diff
@@ -0,0 +1,3 @@
+USEMODULE += sx1276
```
Comment (aabadie) [inline boards/feather-m0-lora/Makefile.dep:1]: Should only be pulled-in when `netdev_default` is used:

```suggestion
ifneq (,$(filter netdev_default,$(USEMODULE)))
  USEMODULE += sx1276
endif
```

File: boards/feather-m0/include/board.h [hunk 2]
```diff
@@ -54,6 +54,20 @@ extern "C" {
 #define ATWINC15X0_PARAM_WAKE_PIN       GPIO_UNDEF
 /** @} */
 
+/**
+ * @name    Configuration for Feather M0 LoRa and the SX1272 module
```
Comment (aabadie) [inline boards/feather-m0/include/board.h:58]: sx1272 or sx1276 ? In the board dependencies it's sx1276 that is used.

**Comment by aabadie** [comment]
Commit message of 549017055fb9e6f3944d63f1b44aa338fc170b78 contains a typo: fether-m0-lora => feather-m0-lora.

File: boards/feather-m0/include/board.h (hunk 2 above)
Comment (miri64) [reply boards/feather-m0/include/board.h:58]: Adafruit doc says sx1276, so probably a typo. Please fix @tperale

**Comment by miri64** [comment]
> Commit message of [5490170](https://github.com/RIOT-OS/RIOT/commit/549017055fb9e6f3944d63f1b44aa338fc170b78) contains a typo: fether-m0-lora => feather-m0-lora.

I assumed this would be fixed when the commits are squashed ;-)

**Comment by miri64** [comment]
Request to squash stands.

File: examples/lua_REPL/Makefile.ci [hunk 3]
```diff
@@ -21,6 +21,7 @@ BOARD_INSUFFICIENT_MEMORY := \
     esp8266-sparkfun-thing \
     feather-m0 \
     feather-m0-wifi \
+    feather-m0-lora \
```
Comment (miri64) [inline examples/lua_REPL/Makefile.ci:24]: Please keep these in alphabetical order.

File: tests/bench_xtimer/Makefile [hunk 4]
```diff
@@ -26,6 +26,7 @@ LOW_MEMORY_BOARDS += \
   derfmega128 \
   feather-m0 \
   feather-m0-wifi \
+  feather-m0-lora \
```
Comment (miri64) [inline tests/bench_xtimer/Makefile:29]: Here too.

**Comment by tperale** [comment]
@miri64 

> ACK. I tested using a bare `feather-m0` (since I don't own the LoRa) version. Flashing `examples/gnrc_networking` or `examples/gnrc_lorawan` works as expected, but of course the application crashes whenever I try to access the network device (since my board does not have one connected). Please squash your commits. I suggest one commit for all modifications in `boards/` as an introductory commit for the `feather-m0-lora` board and one commit for all modifications in `examples` and `tests/` (additions to `{examples,tests}/*/Makefile.ci`; Murdock will reveal more here most likely).

My last modification should be alright. Is there any convention on the squashed commit name ?

**Comment by miri64** [comment]
> My last modification should be alright. Is there any convention on the squashed commit name ?

The commit message should reflect the changes in the commit. It is not necessary to document the ongoing history during the review. E.g.

- `boards/feather-m0-lora: add support`
- `tests / examples: add feather-m0-lora to insufficient memory list`

**Comment by miri64** [comment]
See also our contribution guidelines: https://github.com/RIOT-OS/RIOT/blob/master/CONTRIBUTING.md#commit-conventions

File: boards/feather-m0/doc.txt [hunk 5]
```diff
@@ -65,6 +65,21 @@ and define the required WiFi parameters, for example:
...
+### Using with LoRa module
+
+To enable the LoRa module available on the
+[Feather M0 LoRa](https://learn.adafruit.com/adafruit-feather-m0-radio-with-lora-radio-module)
+variant of the board and automatically be compatible with LoRa applications
+add the `feather-m0-lora` to your project `Makefile`.
+
+```
+BOARD = feather-m0-lora
+USEMODULE += netdev_default
```
Comment (miri64) [inline boards/feather-m0/doc.txt:77]: Let's make this reflective of the WiFi doc as well:

~~~suggestion
To enable the LoRa module available on the
[Feather M0 LoRa](https://learn.adafruit.com/adafruit-feather-m0-radio-with-lora-radio-module)
variant of the board automatically for LoRa applications,
use `feather-m0-lora` as board:

```
make BOARD=feather-m0-lora -C examples/gnrc_lorawan
~~~

**Review by miri64 — CHANGES_REQUESTED** [review]

File: boards/feather-m0/doc.txt (hunk 5 above)
Comment (miri64) [reply boards/feather-m0/doc.txt]: @leandrolanzieri could you confirm, that there are no further parameters needed, e.g. keys and such for LoRa?

Comment (leandrolanzieri) [reply boards/feather-m0/doc.txt]: I'd say that that other settings (such as keys) are out of the scope of the documentation of the board. Many of these parameters depend on the stack being used (GNRC LoRaWAN or Semtech LoraMac). I'd expect the user to check the application README to check how to configure these parameters. So it looks OK to me as proposed.

**Review by miri64 — APPROVED** [review]
Re-ACK. Changes since my last ACK only involved Doc changes, so I don't need to retest (and the `Makefile.dep` change, but that Murdock should be able to catch).

**Comment by miri64** [comment]
And go! Thanks for your contribution @tperale!

**Pull request merged**